src/jit/optimizer.cpp

```diff
@@ -7837,7 +7837,9 @@ GenTree *           Compiler::optIsBoolCond(GenTree *   condBranch,
     if  (opr2->gtOper != GT_CNS_INT)
         return  NULL;
 
-    if  (((unsigned) opr2->gtIntCon.gtIconVal) > 1)
+    ssize_t ival2 = opr2->gtIntCon.gtIconVal;
+
+    if (ival2 != 0 && ival2 != 1)
         return NULL;
```

Comment:
How about adding a new method IsZeroOrOneIntCon() or IsBoolIntCon() on Gentree.cpp to avoid duplication of the same check and also to make the intent is even more clear?

e.g.
bool Gentree::IsBoolIntCon() const
{
   return OperGet() == GT_CNS_INT &&
              (gtIntCon.gtIconVal == 0 || gtIntCon.gtIconVal == 1);
}

Lines 7837 to 7842 can be folded into

if (!opr2->IsBoolIntCon())
    return nullptr;

Similarly lines 7853 to 7859 could be written as
if (op1->gtFlags & GTF_BOOLEAN)
{
...
}
else if (opr1->gtOper == GT_LCL_VAR)
{
....
}
else
{
    isBool = opr1->IsBoolIntCon();
}

I don't feel strongly about this suggestion and will leave it up to you.